pkg/disttask/importinto/proto.go

```diff
@@ -81,27 +86,88 @@ const (
 type MergeSortStepMeta struct {
 	external.BaseExternalMeta
 	// KVGroup is the group name of the sorted kv, either dataKVGroup or index-id.
-	KVGroup               string   `json:"kv-group"`
-	DataFiles             []string `json:"data-files" external:"true"`
-	external.SortedKVMeta `external:"true"`
+	KVGroup                 string   `json:"kv-group"`
+	DataFiles               []string `json:"data-files" external:"true"`
+	external.SortedKVMeta   `external:"true"`
+	RecordedConflictKVCount uint64 `json:"recorded-conflict-kv-count,omitempty"`
 }
 
-// Marshal marshal the merge sort step meta to JSON.
+// Marshal the merge sort step meta to JSON.
 func (m *MergeSortStepMeta) Marshal() ([]byte, error) {
 	return m.BaseExternalMeta.Marshal(m)
 }
 
+// KVGroupConflictInfos is the conflict infos of a kv group.
+type KVGroupConflictInfos struct {
+	ConflictInfos map[string]*common.ConflictInfo `json:"conflict-infos,omitempty"`
+}
+
+func (gci *KVGroupConflictInfos) addDataConflictInfo(other *common.ConflictInfo) {
+	gci.addConflictInfo(dataKVGroup, other)
+}
+
+func (gci *KVGroupConflictInfos) addIndexConflictInfo(indexID int64, other *common.ConflictInfo) {
+	kvGroup := IndexID2KVGroup(indexID)
+	gci.addConflictInfo(kvGroup, other)
+}
+
+func (gci *KVGroupConflictInfos) addConflictInfo(kvGroup string, other *common.ConflictInfo) {
+	if other.Count == 0 {
+		return
+	}
+	if gci.ConflictInfos == nil {
+		gci.ConflictInfos = make(map[string]*common.ConflictInfo, 1)
+	}
+	ci, ok := gci.ConflictInfos[kvGroup]
+	if !ok {
+		ci = &common.ConflictInfo{}
+		gci.ConflictInfos[kvGroup] = ci
+	}
+	ci.Merge(other)
+}
+
+// CollectConflictsStepMeta is the meta of collect conflicts step.
+type CollectConflictsStepMeta struct {
+	external.BaseExternalMeta `json:"base"`
```

Comment:
keep it same as other subtask meta?
```suggestion
	external.BaseExternalMeta
```